test/performance/odp_packet_gen.c

```diff
@@ -1307,34 +1317,46 @@ static inline int send_burst(odp_pktout_queue_t pktout, odp_packet_t pkt[],
 			idx = rand_data[rand_idx++] % pkts_per_pktio;
 		}
 
-		pkt_ref[i] = odp_packet_ref_static(pkt[idx]);
-		if (odp_unlikely(pkt_ref[i] == ODP_PACKET_INVALID)) {
-			num = i;
-			break;
+		if (use_refs) {
+			out_pkt[i] = odp_packet_ref_static(pkt[idx]);
+			if (odp_unlikely(out_pkt[i] == ODP_PACKET_INVALID)) {
+				num = i;
+				break;
+			}
+		} else {
+			out_pkt[i] = pkt[idx];
+			pkt[idx] = ODP_PACKET_INVALID;
```

Comment:
Is this invalidation necessary? Looks like one could simply use `odp_packet_free_multi()` below to free the possible "extra" packets.

Reply:
Yes, `idx` might be random.